tests/test_external_mixin.py

```diff
@@ -377,3 +401,160 @@ def test_restore_state(self, strategy):
         assert strategy._connected is False
         assert strategy._use_template_strategy is False
         strategy.state.restore_state.assert_called_once_with(state_dict)
+
+    @parameterized.expand([
+        [ext_strategy_fixture(LoadForecastExternalStrategy())],
+        [ext_strategy_fixture(PVForecastExternalStrategy())]
+    ])
+    def test_set_energy_forecast(self, strategy):
+        # test successful call of set_energy_forecast:
+        transaction_id = str(uuid.uuid4())
+        arguments = {"transaction_id": transaction_id,
+                     "energy_forecast": {now().format(DATE_TIME_FORMAT): 1}}
+        payload = {"data": json.dumps(arguments)}
+        assert strategy.pending_requests == deque([])
+        strategy.set_energy_forecast(payload)
+        assert len(strategy.pending_requests) > 0
+        energy_forecast_response_channel = f"{strategy.channel_prefix}/" \
+                                           "response/set_energy_forecast"
+        assert (strategy.pending_requests ==
+                deque([IncomingRequest("set_energy_forecast", arguments,
+                                       energy_forecast_response_channel)]))
+
+        # test if set_energy_forecast raises error when provided with wrong payload
+        strategy.redis.publish_json = Mock()
+        strategy.pending_requests = deque([])
+        payload = {"data": json.dumps({"transaction_id": transaction_id})}
+        strategy.set_energy_forecast(payload)
+        strategy.redis.publish_json.assert_called_with(energy_forecast_response_channel,
+                                                       {"command": "set_energy_forecast",
+                                                        "error": "Incorrect "
+                                                                 "_set_energy_forecast request. "
+                                                        "Available parameters: (energy_forecast).",
+                                                        "transaction_id": transaction_id})
+        assert len(strategy.pending_requests) == 0
+
+    @parameterized.expand([
+        [ext_strategy_fixture(LoadForecastExternalStrategy())],
+        [ext_strategy_fixture(PVForecastExternalStrategy())]
+    ])
+    def test_set_energy_measurement(self, strategy):
+        # test successful call of set_energy_measurement:
```

Comment:
```suggestion
        # test successful call of set_energy_measurement.
```
NIT

Reply:
Well, here I want to show that this is done in the following (until the next inline comment appears).